abstract/group.go

```diff
@@ -14,17 +15,27 @@ in which security is based on the Discrete Logarithm assumption,
 and a scalar multiplier in elliptic curve groups.
 */
 type Secret interface {
-	Marshaling
+	RigidMarshaling
+
+	// Produce a human-readable representation of a Secret
+	String() string
 
 	// Equality test for two Secrets derived from the same Group
 	Equal(s2 Secret) bool
 
 	// Set equal to another Secret a
 	Set(a Secret) Secret
 
+	// Set to a big integer value
+	SetInt(v *big.Int) Secret
```

Comment:
I'm not necessarily a fan of overloading the `Secret` interface with this method, when you already added the `SetBytes` method which for me has the same semantic idea: a big.Int is *only* a slice of bytes, both are equivalent. Moreover, this idea of equivalence prevails with the `Secret` becoming a `Scalar`. 
Also, here it makes explicit that a secret *must* be a `big.Int` or at least must deal with `big.Int` where I don't see any reason it should be enforced. Maybe for some weird encryption/signing mechanisms, a scalar won't just *be* a scalar but something more. 
Finally, you don't seem to use it for now ...